internal/rootcoord/dml_channels.go

```diff
@@ -36,40 +42,43 @@ func newDmlChannels(c *Core, chanNamePrefix string, chanNum int64) *dmlChannels
 		core:       c,
 		namePrefix: chanNamePrefix,
 		capacity:   chanNum,
-		refcnt:     sync.Map{},
+		ref:        make([]*dmlRefCnt, chanNum),
 		idx:        atomic.NewInt64(0),
 		pool:       sync.Map{},
 	}
 
-	var i int64
-	for i = 0; i < chanNum; i++ {
-		name := fmt.Sprintf("%s_%d", d.namePrefix, i)
+	for i := int64(0); i < chanNum; i++ {
+		name := getDmlChannelName(d.namePrefix, i)
 		ms, err := c.msFactory.NewMsgStream(c.ctx)
 		if err != nil {
 			log.Error("Failed to add msgstream", zap.String("name", name), zap.Error(err))
 			panic("Failed to add msgstream")
 		}
 		d.pool.Store(name, &ms)
+		d.ref[i] = &dmlRefCnt{
+			mutex: sync.RWMutex{},
+			cnt:   0,
+		}
 	}
 	log.Debug("init dml channels", zap.Int64("num", chanNum))
 	return d
 }
 
 func (d *dmlChannels) GetDmlMsgStreamName() string {
-	cnt := d.idx.Load()
-	name := fmt.Sprintf("%s_%d", d.namePrefix, cnt)
-	d.idx.Store((cnt + 1) % d.capacity)
-	return name
+	cnt := d.idx.Inc()
+	return getDmlChannelName(d.namePrefix, (cnt-1)%d.capacity)
 }
 
 // ListChannels lists all dml channel names
 func (d *dmlChannels) ListChannels() []string {
```

Comment:
We should name it to ListPysical channels?

Reply:
done